packages/perennial/contracts/multiinvoker/MultiInvoker.sol

```diff
@@ -276,6 +276,8 @@ contract MultiInvoker is IMultiInvoker, UInitializable {
      * @param amount Amount of DSU to withdraw from the collateral account
      */
     function _withdrawAndUnwrap(address receiver, IProduct product, UFixed18 amount) internal {
+        // If amount is uint256 max, withdraw the entire balance
+        amount = amount.eq(UFixed18Lib.MAX) ? collateral.collateral(msg.sender, product) : amount;
```

Comment:
Because you're calling this before `collateral.withdrawFrom()` the value is not guaranteed to be synced, so it will be different from the real max - I think you need to pass the `UFixed18Lib.MAX` through, but then not sure how to handle the unwrap properly. Worst case you'll have to call settle() as a first step.

Reply:
Good catch, will have the multiinvoker settle the account if they are withdrawing the full amount